gateway-service-impl/src/main/java/org/hypertrace/gateway/service/GatewayServiceImpl.java

```diff
@@ -61,6 +64,11 @@ public class GatewayServiceImpl extends GatewayServiceGrpc.GatewayServiceImplBas
   private static final String REQUEST_TIMEOUT_CONFIG_KEY = "request.timeout";
   private static final int DEFAULT_REQUEST_TIMEOUT_MILLIS = 10000;
 
+  private Counter serviceResponseErrorCounter;
+  private Counter serviceResponseSuccessCounter;
+  private static final String ERROR_COUNTER_NAME = "hypertrace.gateway.response.errors";
```

Comment:
Shall we have single metrics instead of two with status for fail/success? This way, you can check the call rate too.
e.g
hypertrace_gateway_requests_status {"code"="fail"}
hypertrace_gateway_requests_status {"code"="success"}

Reply:
> Shall we have single metrics instead of two with status for fail/success?

Yes makes sense.

Below looks good:
```
hypertrace.gateway.service.requests.status {"code"="fail"}
hypertrace.gateway.service.requests.status {"code"="success"}
```

> shall we have the same naming convention across services?

Yes my bad, keeping the same naming convention across both the services like - `hypertrace.<servicename>.service.requests.status`

Pushing the above changes.